### DISCUSSION TITLE
compilation error

### DISCUSSION THREAD

**Discussion by sedoy-4** [Hieromon/AutoConnect#570] [Q&A] [answered]
suddenly a compilation error began to appear.
1. IDE delete - setup
2. AutoConnect_library setup v1.4.1
3. Uploading to the board without <AutoConnect.h> happens without problems and the sketch works

I don't know what to do
Help me!
thanks

C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.cpp: In member function 'void AutoConnectTicker::start()':
C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.cpp:32:83: error: no matching function for call to 'Ticker::attach_ms<AutoConnectTicker*>(uint32_t&, void (&)(AutoConnectTicker*), AutoConnectTicker*)'
   32 |   _period.attach_ms<AutoConnectTicker*>(_cycle, AutoConnectTicker::_onPeriod, this);
      |                                                                                   ^
In file included from C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.h:19,
                 from C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.cpp:12:
C:\Users\dns\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.1.1\libraries\Ticker\src/Ticker.h:116:10: note: candidate: 'template<class Func, class Arg> void Ticker::attach_ms(uint32_t, Func, Arg)'
  116 |     void attach_ms(uint32_t milliseconds, Func func, Arg arg)
      |          ^~~~~~~~~
C:\Users\dns\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.1.1\libraries\Ticker\src/Ticker.h:116:10: note:   template argument deduction/substitution failed:
C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.cpp:32:68: note:   cannot convert 'AutoConnectTicker::_onPeriod' (type 'void(AutoConnectTicker*)') to type 'AutoConnectTicker*'
   32 |   _period.attach_ms<AutoConnectTicker*>(_cycle, AutoConnectTicker::_onPeriod, this);
      |                                                 ~~~~~~~~~~~~~~~~~~~^~~~~~~~~
C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.cpp: In static member function 'static void AutoConnectTicker::_onPeriod(AutoConnectTicker*)':
C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.cpp:45:81: error: no matching function for call to 'Ticker::once_ms<AutoConnectTicker*>(uint32_t&, void (&)(AutoConnectTicker*), AutoConnectTicker*&)'
   45 |   t->_pulse.once_ms<AutoConnectTicker*>(t->_duty, AutoConnectTicker::_onPulse, t);
      |                                                                                 ^
In file included from C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.h:19,
                 from C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.cpp:12:
C:\Users\dns\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.1.1\libraries\Ticker\src/Ticker.h:160:10: note: candidate: 'template<class Func, class Arg> void Ticker::once_ms(uint32_t, Func, Arg)'
  160 |     void once_ms(uint32_t milliseconds, Func func, Arg arg)
      |          ^~~~~~~
C:\Users\dns\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\3.1.1\libraries\Ticker\src/Ticker.h:160:10: note:   template argument deduction/substitution failed:
C:\Users\dns\Documents\Arduino\libraries\AutoConnect\src\AutoConnectTicker.cpp:45:70: note:   cannot convert 'AutoConnectTicker::_onPulse' (type 'void(AutoConnectTicker*)') to type 'AutoConnectTicker*'
   45 |   t->_pulse.once_ms<AutoConnectTicker*>(t->_duty, AutoConnectTicker::_onPulse, t);

**Comment by sedoy-4** [comment]
Excuse me: IDE delete - installed

**Comment by Hieromon** [comment]
@sedoy-4 
AutoConnect does not yet support ESP8266 Arduino Core 3.1.0 or later. I'm are currently working on it.
See an issue #567 for a tentative workaround.

Apart from this interim workaround, you can avoid the error by applying [ESP8266 Arduino Core Release 3.0.2](https://github.com/esp8266/Arduino/releases/tag/3.0.2). Select ESP8266 version 3.0.2 in the board manager of the Arduino IDE and install it.

**Reply by sedoy-4** [reply]
thank you very much!!! :)

**Comment by sedoy-4** [comment]
Installed Auto Connect V 1.3.4, ESP8266 core V 2.7.7 and everything worked!!! Thanks a lot for the hint! Wonderful library! Good luck and prosperity to you!!!

**Comment by Hieromon** [comment, marked as answer]
Fixed #575
